smithy-model/src/test/java/software/amazon/smithy/model/knowledge/ServiceIndexTest.java

```diff
@@ -80,10 +83,11 @@ public void getsAuthSchemesOfServiceWithNoAuthTrait() {
         Map<ShapeId, Trait> auth = serviceIndex.getEffectiveAuthSchemes(
                 ShapeId.from("smithy.example#ServiceWithNoAuthTrait"));
 
-        assertThat(auth.keySet(), hasSize(3));
-        assertThat(auth, hasKey(HttpBasicAuthTrait.ID));
-        assertThat(auth, hasKey(HttpDigestAuthTrait.ID));
-        assertThat(auth, hasKey(HttpBearerAuthTrait.ID));
+        List<ShapeId> ids = new ArrayList<>(auth.keySet());
+        assertThat(ids, hasSize(3));
+        assertThat(ids.get(0), equalTo(HttpBasicAuthTrait.ID));
```

Comment:
`getsAuthSchemesOfServiceWithAuthTrait` below also should assert the order instead of just hasKey.